# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1292690614 28800
#      Sat Dec 18 08:43:34 2010 -0800
# Node ID 7ca273af4309616306c9d6fbecec436339e160b2
# Parent  934ed3e07542424c54b5faf4adcc69ed8e734aea
Deprecate sphcat and spvcat functions.

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-12-18  Rik  <octave@nomad.inbox5.com>
+
+	* NEWS: Deprecate sphcat and spvcat functions.
+
 2010-12-17  Rik  <octave@nomad.inbox5.com>
 
 	* README.MacOS: New file documenting build instructions for Mac OSX.
 
 2010-12-09  Kai Habel  <kai.habel@gmx.de>
 
 	* NEWS: Add pie3 to new functions list.
 
diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -388,27 +388,27 @@ Summary of important user-visible change
       gamma_rnd        normal_rnd           wiener_rnd      
       geometric_cdf    pascal_cdf
       geometric_inv    pascal_inv
 
     The following functions were deprecated in Octave 3.2 and will
     be removed from Octave 3.6 (or whatever version is the second major
     release after 3.2):
 
-      create_set          spcholinv  spmax
-      dmult               spcumprod  spmin
-      iscommand           spcumsum   spprod
-      israwcommand        spdet      spqr
-      lchol               spdiag     spsum
-      loadimage           spfind     spsumsq
-      mark_as_command     spinv      str2mat
-      mark_as_rawcommand  spkron     unmark_command
-      spatan2             splchol    unmark_rawcommand
-      spchol              split
-      spchol2inv          splu
+      create_set          spcholinv    splu   
+      dmult               spcumprod    spmax
+      iscommand           spcumsum     spmin
+      israwcommand        spdet        spprod
+      lchol               spdiag       spqr
+      loadimage           spfind       spsum
+      mark_as_command     sphcat       spsumsq
+      mark_as_rawcommand  spinv        spvcat 
+      spatan2             spkron       str2mat
+      spchol              splchol      unmark_command
+      spchol2inv          split        unmark_rawcommand
 
     The following functions have been deprecated in Octave 3.4 and will
     be removed from Octave 3.8 (or whatever version is the second major
     release after 3.4):
 
       autocor  betai    dispatch  gammai     values
       autocov  cellidx  fstat     is_global
 
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2010-10-28  Rik  <octave@nomad.inbox5.com>
+
+	* deprecated/module.mk, sparse/module.mk, deprecated/sphcat.m,
+	deprecated/spvcat.m: Deprecate sphcat and spvcat functions.
+
 2010-12-17  John W. Eaton  <jwe@octave.org>
 
 	* plot/comet3.m: New function.
 	* plot/module.mk: Add comet3.m to the list
 	* miscellaneous/unimplemented.m: Remove comet3 from the list.
 
 2010-12-17  John W. Eaton  <jwe@octave.org>
 
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -18,35 +18,37 @@ deprecated_FCN_FILES = \
   deprecated/israwcommand.m \
   deprecated/isstr.m \
   deprecated/lchol.m \
   deprecated/loadimage.m \
   deprecated/mark_as_command.m \
   deprecated/mark_as_rawcommand.m \
   deprecated/setstr.m \
   deprecated/spatan2.m \
-  deprecated/spchol.m \
   deprecated/spchol2inv.m \
   deprecated/spcholinv.m \
+  deprecated/spchol.m \
   deprecated/spcumprod.m \
   deprecated/spcumsum.m \
   deprecated/spdet.m \
   deprecated/spdiag.m \
   deprecated/spfind.m \
+  deprecated/sphcat.m \
   deprecated/spinv.m \
   deprecated/spkron.m \
   deprecated/splchol.m \
   deprecated/split.m \
   deprecated/splu.m \
   deprecated/spmax.m \
   deprecated/spmin.m \
   deprecated/spprod.m \
   deprecated/spqr.m \
   deprecated/spsum.m \
   deprecated/spsumsq.m \
+  deprecated/spvcat.m \
   deprecated/str2mat.m \
   deprecated/unmark_command.m \
   deprecated/unmark_rawcommand.m \
   deprecated/values.m \
   deprecated/weibcdf.m \
   deprecated/weibinv.m \
   deprecated/weibpdf.m \
   deprecated/weibrnd.m
diff --git a/scripts/deprecated/sphcat.m b/scripts/deprecated/sphcat.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/sphcat.m
@@ -0,0 +1,35 @@
+## Copyright (C) 2004, 2005, 2006, 2007, 2009 David Bateman and Andy Adler
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {@var{y} =} sphcat (@var{a1}, @var{a2}, @dots{}, @var{aN})
+## Return the horizontal concatenation of sparse matrices.  This function
+## is obselete and @code{horzcat} should be used instead.
+## @seealso {horzcat, spvcat, vertcat, cat}
+## @end deftypefn
+
+function y = sphcat (varargin)
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "sphcat is obsolete and will be removed from a future version of Octave; please use horzcat instead");
+  endif
+
+  y = horzcat (varargin{:});
+endfunction
diff --git a/scripts/deprecated/spvcat.m b/scripts/deprecated/spvcat.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/spvcat.m
@@ -0,0 +1,35 @@
+## Copyright (C) 2004, 2005, 2006, 2007, 2009 David Bateman and Andy Adler
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Function File} {@var{y} =} spvcat (@var{a1}, @var{a2}, @dots{}, @var{aN})
+## Return the vertical concatenation of sparse matrices.  This function
+## is obselete and @code{vertcat} should be used instead.
+## @seealso{vertcat, sphcat, horzcat, cat}
+## @end deftypefn
+
+function y = spvcat (varargin)
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "spvcat is obsolete and will be removed from a future version of Octave; please use vertcat instead");
+  endif
+
+  y = vertcat (varargin{:});
+endfunction
diff --git a/scripts/sparse/module.mk b/scripts/sparse/module.mk
--- a/scripts/sparse/module.mk
+++ b/scripts/sparse/module.mk
@@ -9,23 +9,21 @@ sparse_FCN_FILES = \
   sparse/nonzeros.m \
   sparse/pcg.m \
   sparse/pcr.m \
   sparse/spaugment.m \
   sparse/spconvert.m \
   sparse/spdiags.m \
   sparse/speye.m \
   sparse/spfun.m \
-  sparse/sphcat.m \
   sparse/spones.m \
   sparse/sprand.m \
   sparse/sprandn.m \
   sparse/sprandsym.m \
   sparse/spstats.m \
-  sparse/spvcat.m \
   sparse/spy.m \
   sparse/svds.m \
   sparse/treelayout.m \
   sparse/treeplot.m
 
 FCN_FILES += $(sparse_FCN_FILES)
 
 PKG_ADD_FILES += sparse/PKG_ADD
diff --git a/scripts/sparse/sphcat.m b/scripts/sparse/sphcat.m
deleted file mode 100644
--- a/scripts/sparse/sphcat.m
+++ /dev/null
@@ -1,36 +0,0 @@
-## Copyright (C) 2004, 2005, 2006, 2007, 2009 David Bateman and Andy Adler
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or (at
-## your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-## General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn {Function File} {@var{y} =} sphcat (@var{a1}, @var{a2}, @dots{}, @var{aN})
-## Return the horizontal concatenation of sparse matrices.  This function
-## is obselete and @code{horzcat} should be used.
-## @seealso {spvcat, vertcat, horzcat, cat}
-## @end deftypefn
-
-function y = sphcat (varargin)
-
-  persistent sphcat_warned = false;
-
-  if (! sphcat_warned)
-    sphcat_warned = true;
-    warning ("sphcat: This function is depreciated.  Use horzcat instead");
-  endif
-
-  y = horzcat (varargin{:});
-endfunction
diff --git a/scripts/sparse/spvcat.m b/scripts/sparse/spvcat.m
deleted file mode 100644
--- a/scripts/sparse/spvcat.m
+++ /dev/null
@@ -1,36 +0,0 @@
-## Copyright (C) 2004, 2005, 2006, 2007, 2009 David Bateman and Andy Adler
-##
-## This file is part of Octave.
-##
-## Octave is free software; you can redistribute it and/or modify it
-## under the terms of the GNU General Public License as published by
-## the Free Software Foundation; either version 3 of the License, or (at
-## your option) any later version.
-##
-## Octave is distributed in the hope that it will be useful, but
-## WITHOUT ANY WARRANTY; without even the implied warranty of
-## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-## General Public License for more details.
-##
-## You should have received a copy of the GNU General Public License
-## along with Octave; see the file COPYING.  If not, see
-## <http://www.gnu.org/licenses/>.
-
-## -*- texinfo -*-
-## @deftypefn {Function File} {@var{y} =} spvcat (@var{a1}, @var{a2}, @dots{}, @var{aN})
-## Return the vertical concatenation of sparse matrices.  This function
-## is obselete and @code{vertcat} should be used
-## @seealso{sphcat, vertcat, horzcat, cat}
-## @end deftypefn
-
-function y = spvcat (varargin)
-
-  persistent spvcat_warned = false;
-
-  if (! spvcat_warned)
-    spvcat_warned = true;
-    warning ("spvcat: This function is depreciated. Use vertcat instead");
-  endif
-
-  y = vertcat (varargin{:});
-endfunction
